probe-rs/src/debug/unit_info.rs

```diff
@@ -440,6 +460,523 @@ impl UnitInfo {
+                            } else {
+                                child_variable.set_value(VariableValue::Error(
+                                    "Unimplemented: Base type without a name".to_string(),
+                                ));
+                            }
+
+                            let location = self.process_memory_location_type_info(
+                                debug_info,
+                                &attributes_entry,
+                                &parent_variable.memory_location,
+                                memory,
+                                frame_info,
+                            )?;
+
+                            child_variable.memory_location = location.clone();
+
+                            if let VariableLocation::Value(value) = location {
+                                child_variable.set_value(value);
+                            }
+                        }
+
+                        TypeKind::Pointer {
+                            byte_size,
+                            pointed_to,
+                        } => {
+                            child_variable.byte_size = Some(*byte_size);
+
+                            child_variable.type_name =
+                                VariableType::Pointer(type_info.name().map(|s| s.to_string()));
+
+                            let location = self.process_memory_location_type_info(
+                                debug_info,
+                                &attributes_entry,
+                                &parent_variable.memory_location,
+                                memory,
+                                frame_info,
+                            )?;
+
+                            child_variable.memory_location = location;
+
+                            child_variable.variable_node_type = VariableNodeType::ReferenceOffset(
+                                pointed_to
+                                    .offset()
+                                    .to_unit_offset(&self.unit.header)
+                                    .unwrap(),
+                            );
+                        }
+
+                        TypeKind::Struct { byte_size, members } => {
+                            child_variable.byte_size = *byte_size;
+
+                            let name = type_info.name().unwrap_or("<unnamed>");
+
+                            child_variable.type_name = VariableType::Struct(name.to_string());
+
+                            let location = self.process_memory_location_type_info(
+                                debug_info,
+                                &attributes_entry,
+                                &parent_variable.memory_location,
+                                memory,
+                                frame_info,
+                            )?;
+
+                            let resolve_children = name.starts_with("&str")
```

Comment:
Can we move these to some lang-specific function? Or at least could you add a todo to these to revisit later?

Reply:
FYI I've done this as part of https://github.com/probe-rs/probe-rs/pull/2199